deepfence_worker/controls/controls.go

```diff
@@ -54,101 +55,72 @@ func init() {
 	if err != nil {
 		return err
 	}
 
 	// for secret scan
 	err = RegisterControl(ctl.StartSecretScan,
-		func(namespace string, req ctl.StartSecretScanRequest) error {
-			metadata := map[string]string{directory.NamespaceKey: namespace}
-			log.Info().Msgf("payload: %+v", req.BinArgs)
-			data, err := json.Marshal(req.BinArgs)
-			if err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			if err := utils.PublishNewJob(pub, metadata, sdkUtils.SecretScanTask, data); err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			return nil
-		})
+		GetRegisterControlFunc[ctl.StartSecretScanRequest](pub,
+			sdkUtils.SecretScanTask))
 	if err != nil {
 		return err
 	}
 
-	err = RegisterControl(ctl.StopSecretScan,
-		func(namespace string, req ctl.StopSecretScanRequest) error {
-			metadata := map[string]string{directory.NamespaceKey: namespace}
-			log.Info().Msgf("StopSecretScan payload: %+v", req.BinArgs)
-			data, err := json.Marshal(req.BinArgs)
-			if err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			if err := utils.PublishNewJob(pub, metadata, sdkUtils.StopSecretScanTask, data); err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			return nil
-		})
+	// for malware scan
+	err = RegisterControl(ctl.StartMalwareScan,
+		GetRegisterControlFunc[ctl.StartMalwareScanRequest](pub,
+			sdkUtils.MalwareScanTask))
 	if err != nil {
 		return err
 	}
 
-	// for malware scan
-	err = RegisterControl(ctl.StartMalwareScan,
-		func(namespace string, req ctl.StartMalwareScanRequest) error {
-			metadata := map[string]string{directory.NamespaceKey: namespace}
-			log.Info().Msgf("payload: %+v", req.BinArgs)
-			data, err := json.Marshal(req.BinArgs)
-			if err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			if err := utils.PublishNewJob(pub, metadata, sdkUtils.MalwareScanTask, data); err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			return nil
-		})
+	err = RegisterControl(ctl.StopSecretScan,
+		GetRegisterControlFunc[ctl.StopSecretScanRequest](pub,
+			sdkUtils.StopSecretScanTask))
 	if err != nil {
 		return err
 	}
 
 	err = RegisterControl(ctl.StopMalwareScan,
-		func(namespace string, req ctl.StopMalwareScanRequest) error {
-			metadata := map[string]string{directory.NamespaceKey: namespace}
-			log.Info().Msgf("StopMalwareScan payload: %+v", req.BinArgs)
-			data, err := json.Marshal(req.BinArgs)
-			if err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			if err := utils.PublishNewJob(pub, metadata, sdkUtils.StopMalwareScanTask, data); err != nil {
-				log.Error().Msg(err.Error())
-				return err
-			}
-			return nil
-		})
+		GetRegisterControlFunc[ctl.StopMalwareScanRequest](pub,
+			sdkUtils.StopMalwareScanTask))
+	if err != nil {
+		return err
+	}
+
+	err = RegisterControl(ctl.StopVulnerabilityScan,
+		GetRegisterControlFunc[ctl.StopVulnerabilityScanRequest](pub,
+			sdkUtils.StopVulnerabilityScanTask))
```

Comment:
nit: for consistency, let's put stopVuln scan before secret & malware, to keep them in the same order as the starts